Source/WebKit/GPUProcess/cocoa/GPUProcessCocoa.mm

```diff
@@ -89,6 +94,33 @@
 }
 #endif
 
+void GPUProcess::platformInitializeGPUProcess(GPUProcessCreationParameters& parameters)
+{
+#if PLATFORM(MAC)
+    auto launchServicesExtension = SandboxExtension::create(WTFMove(parameters.launchServicesExtensionHandle));
+    if (launchServicesExtension) {
+        bool ok = launchServicesExtension->consume();
+        ASSERT_UNUSED(ok, ok);
+    }
+
+    // It is important to check in with launch services before setting the process name.
+    launchServicesCheckIn();
+
+    // Update process name while holding the Launch Services sandbox extension
+    updateProcessName();
+
+    // Close connection to launch services.
+#if HAVE(HAVE_LS_SERVER_CONNECTION_STATUS_RELEASE_NOTIFICATIONS_MASK)
+    _LSSetApplicationLaunchServicesServerConnectionStatus(kLSServerConnectionStatusDoNotConnectToServerMask | kLSServerConnectionStatusReleaseNotificationsMask, nullptr);
+#else
+    _LSSetApplicationLaunchServicesServerConnectionStatus(kLSServerConnectionStatusDoNotConnectToServerMask, nullptr);
+#endif
```

Comment:
Nit: this could be moved into the if (launchServicesExtension) clause, since it will not work without the extension.

Reply:
Does it hurt to keep as is?

Reply:
Keeping as-is is fine :)